cpp/src/arrow/util/parsing.cc

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/util/parsing.h"
+#include "arrow/util/double_conversion.h"
+
+namespace arrow {
+namespace internal {
+
+struct StringToFloatConverter::Impl {
+  Impl()
+      : main_converter_(flags_, main_junk_value_, main_junk_value_, "inf", "nan"),
+        fallback_converter_(flags_, fallback_junk_value_, fallback_junk_value_, "inf",
+                            "nan") {}
+
+  // This is only supported in double-conversion 3.1+
+#ifdef DOUBLE_CONVERSION_HAS_CASE_INSENSIBILITY
```

Comment:
You don't need this anymore if you vendor it.

Reply:
Hmm, true.